src/clj/harja/palvelin/raportointi/raportit/vastaanottotarkastus.sql

```diff
@@ -101,25 +105,64 @@ WHERE ((:urakka::INTEGER IS NULL AND u.urakkanro IS NOT NULL)
       AND s.poistettu IS NOT TRUE
 ORDER BY s.perintapvm DESC;
 
--- name: pkluokkien-kustannukset-hallintayksikoittain
+-- name: pkluokkien-kustannukset-urakoittain
 -- Päätellään päällystyskohteen pk-luokka laskemalla alikohteiden (yllapitokohdeosa) pituudet ja ottamalla se
 -- pk-luokka, jonka pituus on suurin
-SELECT DISTINCT ON (CONCAT(u.id, ' ', u.nimi)) CONCAT(u.id, ' ', u.nimi)          AS nimi,
-                                               o.id                               AS "hallintayksikko_id",
-                                               o.nimi                             AS "hallintayksikko_nimi",
-                                               ypk.pkluokka,
-                                               SUM(ypkk.sopimuksen_mukaiset_tyot) AS "sopimuksen-mukaiset-tyot",
-                                               SUM(ypkk.maaramuutokset)           AS "maaramuutokset",
-                                               SUM(ypkk.arvonvahennykset)         AS "arvonvahennykset",
-                                               SUM(ypkk.toteutunut_hinta)         AS "toteutunut-hinta",
-                                               SUM(ypkk.bitumi_indeksi)           AS "bitumi-indeksi",
-                                               SUM(ypkk.kaasuindeksi)             AS "kaasuindeksi",
-                                               SUM(ypkk.maku_paallysteet)         AS "maku-paallysteet"
+  WITH sakot AS (SELECT SUM(-(s.maara)) AS sakot,
+                        u.id          AS urakkaid
+                   FROM urakka u
+                            JOIN organisaatio o ON u.hallintayksikko = o.id
+                            JOIN laatupoikkeama lp ON lp.urakka = u.id AND lp.poistettu IS NOT TRUE
+                            JOIN sanktio s ON s.laatupoikkeama = lp.id AND s.poistettu IS NOT TRUE
+                  WHERE (SELECT EXTRACT(YEAR FROM lp.aika)) = :vuosi
+                    AND (1::INTEGER IS NULL OR u.hallintayksikko = 1)
```

Comment:
> 1::INTEGER IS NULL

Onko tämä jokin placeholder? Koska tuo ei tee mitään, se on aina false. 
Samakun kysyy IF TRUE IS FALSE, joka on aina false.

Reply:
Tää oli paha. Hyvä, että huomasit. Tuo meinaa siis sitä, että vaikka valitsee Lapin, niin se antaa aina Uudenmaan.